src/main/java/io/kokuwa/maven/helm/UploadMojo.java

```diff
@@ -167,6 +203,39 @@ private void uploadSingle(Path chart) throws MojoExecutionException, IOException
 		connection.disconnect();
 	}
 
+	private boolean verifyUpload(Path chartPath) throws MojoExecutionException {
+		ObjectMapper MAPPER = new YAMLMapper()
+				.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+		String chartName;
+		try {
+			chartName = MAPPER.readValue(chartPath.toFile(), HelmChart.class).getName();
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read chart from " + chartPath, e);
+		}
+
+		long startTime = System.currentTimeMillis();
+		long timeoutMillis = uploadVerificationTimeout * 1000;
```

Comment:
A: Super minor and mostly stylistic, but I kind of like something akin to `cutoff = startTime + timeout` and then the if check is a little more readable with `System.currentTimeMillis() < cutoff`